projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Segments.php

```diff
@@ -232,10 +232,8 @@ public function getSegments($ownerID=-1,$perPage=10,$page=0,$withConditions=fals
                         $page = (int)$page-1;
                         if ($page < 0) $page = 0;
 
-                        // check params realistic
-                        // todo, for now, brute pass
-                        $orderByStr .= ' LIMIT '.(int)$page.','.(int)$perPage;
-
+						// phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+						$orderByStr .= ' LIMIT ' . ( (int) $page * (int) $perPage ) . ',' . (int) $perPage;
```

Comment:
Personally I like to use sprintf in cases like this. E.g. `$orderByStr .= sprintf( ' LIMIT %d,%d ', (int) $page * (int) $perPage, (int) $perPage )`. But this works.

Reply:
Yes, it's better for readability. I directly fixed the line without thinking about improving it. Another method I like is to use `strtr` function for SQL queries because you can name the parameters, so the code will be clearer:

```
$orderByStr .= strtr( ' LIMIT :offset, :perPage', array(
   ':offset'   => (int) $page * (int) $perPage,
   ':perPage'  => (int) $perPage
) );
```